src/Collector/LoggerInterfaceProxy.php

```diff
@@ -50,49 +50,58 @@ public function critical(string|Stringable $message, array $context = []): void
 
     public function error(string|Stringable $message, array $context = []): void
     {
-        [$callStack] = debug_backtrace();
+        $callStack = $this->getCallStack();
 
         $this->collector->collect(LogLevel::ERROR, $message, $context, $callStack['file'] . ':' . $callStack['line']);
         $this->logger->error($message, $context);
     }
 
     public function warning(string|Stringable $message, array $context = []): void
     {
-        [$callStack] = debug_backtrace();
+        $callStack = $this->getCallStack();
 
         $this->collector->collect(LogLevel::WARNING, $message, $context, $callStack['file'] . ':' . $callStack['line']);
         $this->logger->warning($message, $context);
     }
 
     public function notice(string|Stringable $message, array $context = []): void
     {
-        [$callStack] = debug_backtrace();
+        $callStack = $this->getCallStack();
 
         $this->collector->collect(LogLevel::NOTICE, $message, $context, $callStack['file'] . ':' . $callStack['line']);
         $this->logger->notice($message, $context);
     }
 
     public function info(string|Stringable $message, array $context = []): void
     {
-        [$callStack] = debug_backtrace();
+        $callStack = $this->getCallStack();
 
         $this->collector->collect(LogLevel::INFO, $message, $context, $callStack['file'] . ':' . $callStack['line']);
         $this->logger->info($message, $context);
     }
 
     public function debug(string|Stringable $message, array $context = []): void
     {
-        [$callStack] = debug_backtrace();
+        $callStack = $this->getCallStack();
 
         $this->collector->collect(LogLevel::DEBUG, $message, $context, $callStack['file'] . ':' . $callStack['line']);
         $this->logger->debug($message, $context);
     }
 
     public function log(mixed $level, string|Stringable $message, array $context = []): void
     {
-        [$callStack] = debug_backtrace();
+        $callStack = $this->getCallStack();
 
         $this->collector->collect($level, $message, $context, $callStack['file'] . ':' . $callStack['line']);
         $this->logger->log($level, $message, $context);
     }
+
+    /**
+     * @psalm-return array{file: string, line: int}
+     */
+    private function getCallStack(): array
+    {
+        /** @psalm-var array{file: string, line: int} */
+        return debug_backtrace()[0];
```

Comment:
You should add +1 because stack has changed

Reply:
Why? It change current behaviour. 

In PR replaced `[$callstack] = debug_backtrace()` to `$callstack = debug_backtrace()[0]`. It is same, isn't it?

Reply:
Yes, but you have moved the function call into another function

Reply:
Oh. I understand. 
I will fix it and try create test.

Reply:
Fixed